src/templates/LectureView/VideoPlayer.js

```diff
@@ -0,0 +1,68 @@
+import React, { useContext } from 'react'
+import { useTranslation } from 'react-i18next'
+import ReactPlayer from 'react-player'
+import PropTypes from 'prop-types'
+
+import { toast } from 'react-toastify'
+import { AuthContext } from '../../contexts/AuthContext'
+import { addProfile } from '../../services/api'
+
+const VideoPlayer = ({ url, lectureStrapiId, courseStrapiId }) => {
+  const DEFENITION_OF_COMPLETED = 0.8
+  const { currentUser, isLoggedIn, setCurrentUser } = useContext(AuthContext)
+  const { t } = useTranslation()
+
+  const lectures =
+    isLoggedIn &&
+    currentUser &&
+    currentUser.profile &&
+    currentUser.profile.completedlectures
+      ? currentUser.profile.completedlectures
+      : {}
+
+  const updateCompletedLectures = async updatedLectures => {
+    try {
+      const response = await addProfile({ completedlectures: updatedLectures })
+      setCurrentUser({ ...currentUser, profile: response })
+      toast.success(t('lectureCompletedMsg'))
+    } catch (err) {
+      console.log(err)
```

Comment:
We do not have error handling for now, we can just add FIXME 
```suggestion
      // FIXME: error handling
      // console.log(err)
```

Reply:
All done